openfeature/provider/__init__.py

```diff
@@ -7,6 +8,14 @@
 from .metadata import Metadata
 
 
+class ProviderStatus(Enum):
+    NOT_READY = "NOT_READY"
+    READY = "READY"
+    ERROR = "ERROR"
+    STALE = "STALE"
+    FATAL = "FATAL"
```

Comment:
Consider adding the `FATAL` error code as well, relevant to my other comment [here](https://github.com/open-feature/python-sdk/pull/288/files#r1515105556).